test_regress/t/t_tri_top_en_out.cpp

```diff
@@ -0,0 +1,73 @@
+// DESCRIPTION: Verilator: Verilog Test module, C driver code
+//
+// This file ONLY is placed under the Creative Commons Public Domain, for
+// any use, without warranty, 2024 by Paul Wright.
+// SPDX-License-Identifier: CC0-1.0
+
+#include "verilated.h"
+
+#include "Vt_tri_top_en_out.h"
+
+#include <cstdio>
+
+//======================
+
+int main(int argc, char** argv, char**) {
+    // Setup context, defaults, and parse command line
+    Verilated::debug(0);
+    const std::unique_ptr<VerilatedContext> contextp{new VerilatedContext};
+    contextp->commandArgs(argc, argv);
+
+    // Construct the Verilated model, from Vtop.h generated from Verilating
+    const std::unique_ptr<Vt_tri_top_en_out> topp{new Vt_tri_top_en_out{contextp.get()}};
+
+    // Initial input
+    topp->drv_en = 1;
+    topp->single_bit_io = 0;
+    topp->bidir_single_bit_io = 0;
+    topp->bus_64_io = 0;
+    topp->bidir_bus_64_io = 0;
+    topp->bus_128_io[0] = 0;
+    topp->bus_128_io[1] = 0;
+    topp->bus_128_io[2] = 0;
+    topp->bus_128_io[3] = 0;
+    topp->bidir_bus_128_io[0] = 0;
+    topp->bidir_bus_128_io[1] = 0;
+    topp->bidir_bus_128_io[2] = 0;
+    topp->bidir_bus_128_io[3] = 0;
+    topp->sub_io = 0;
+
+    // Simulate until $finish
+    while (!contextp->gotFinish()) {
+        // Evaluate model
+        topp->eval();
+
+        // Advance time (to scheduled events)
+        if (!topp->eventsPending()) break;
+        contextp->time(topp->nextTimeSlot());
+
+        // We want to check that the __en and __out signals can be accessed
+
+        printf("bidir_single_bit_io__en = %x\n", topp->bidir_single_bit_io__en);
+        printf("bidir_bus_64_io__en = %x\n", (unsigned int)topp->bidir_bus_64_io__en);
+        printf("bidir_bus_128_io__en = %x,%x,%x,%x\n", topp->bidir_bus_128_io[3],
+               topp->bidir_bus_128_io[2], topp->bidir_bus_128_io[1], topp->bidir_bus_128_io[0]);
+        printf("sub_io__en = %x\n", topp->sub_io__en);
+
+        printf("bidir_single_bit_io = %x\n", topp->bidir_single_bit_io__out);
+        printf("bidir_bus_64_io = %x\n", (unsigned int)topp->bidir_bus_64_io__out);
+        printf("bidir_bus_128_io = %x,%x,%x,%x\n", topp->bidir_bus_128_io__out[3],
+               topp->bidir_bus_128_io__out[2], topp->bidir_bus_128_io__out[1],
+               topp->bidir_bus_128_io__out[0]);
+        printf("sub_io = %x\n", topp->sub_io__out);
```

Comment:
Please assert all outputs are correct values.  Make sure the tests cover a mix of 0/1/Z on separate bits in a bus.

Reply:
I have updated t_tri_top_en_out.v to check the state driven from verilog (passes)
t_tri_top_en_out.cpp now checks the __en outputs, this fails. I'll need to debug.
I also need to drive in on the undriven inouts and confirm that the values appear on __out.

Reply:
Done for 64-bit and 128 bit bus widths